test/testframe.c

```diff
@@ -193,6 +195,28 @@ TestInit(const char *ProgName, void (*TestPrivateUsage)(FILE *stream),
             test_path_prefix = "";
     }
 
+    /* Get maximum number of test threads from environment, if set */
+    if ((env_var = getenv(HDF5_TEST_MAX_NUM_THREADS))) {
```

Comment:
Added an environment variable so that the CI can easily adjust the number of threads to use since dynamically specifying command-line options is complicated